src/IssueViz.UnitTests/Editor/SingletonDisposableTaggerManagerTests.cs

```diff
@@ -0,0 +1,156 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Tagging;
+using Moq;
+using static SonarLint.VisualStudio.IssueVisualization.UnitTests.Editor.Common.TaggerTestHelper;
+using TypedSingleton = SonarLint.VisualStudio.IssueVisualization.Editor.SingletonDisposableTaggerManager<SonarLint.VisualStudio.IssueVisualization.UnitTests.Editor.SingletonDisposableTaggerManagerTests.IDummyTagType>;
```

Comment:
This is the longest 'using' alias I've ever seen. The singleton manager has generics and nested delegates so I added this alias to make the test code easier to read. Unfortunately, all of the paths in the alias have to be fully qualified, which is why it is so long.

Reply:
🤯